### PR TITLE
Test ViewModel(s)

### PR DESCRIPTION
**Pull request by AnonymousWalker**
This change is [Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434)

### CODE REVIEW

File: jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt
Comment (AnonymousWalker) [inline jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt:50]: Explanation: displays all result when no box is checked.

Previously:
image

**Review by mXaln — CHANGES_REQUESTED** [review]
> Reviewed 1 of 4 files at r1, 1 of 1 files at r4, 1 of 4 files at r7, 1 of 2 files at r8, 8 of 8 files at r10.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434)* status: 11 of 12 files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker and @mXaln)

___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt, line 50 at r8](https://reviewable.io/reviews/bible-translation-tools/orature/434#-Mq-ikLO_R3Qbjub_YfI-r8-50:-MqZcArQ1o9CKXis9Ft9:b-aejgpn) ([raw file](https://github.com/bible-translation-tools/orature/blob/165aff6ffb5305f2bdfdebd4e0dfe6b2fe6714b1/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt#L50)):*
<details><summary><i>Previously, AnonymousWalker (Tony Anh Tran) wrote…</i></summary><blockquote>

Explanation: displays all result when no box is checked.

Previously:
image

</blockquote></details>

I disagree. All checkboxes selected should not  be the same as no checkboses selected.
___
*[jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModelTest.kt, line 107 at r10](https://reviewable.io/reviews/bible-translation-tools/orature/434#-MqZcb81EZR4_JC-Z8Bp:-MqZcb829oc82N2rEIFd:b-5daa89) ([raw file](https://github.com/bible-translation-tools/orature/blob/8258cb4802dd99b8e74b7b732c30f8a1651f9cbe/jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModelTest.kt#L107)):*
> ```Kotlin
> 
>         verify(mockSettingsVM).refreshPlugins()
>         verify(mockWorkbook)
> ```

Perhaps there should be a method here. ```verify(mockWorkbook).someMethod()```?

**Comment by AnonymousWalker** [comment]
___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt, line 50 at r8](https://reviewable.io/reviews/bible-translation-tools/orature/434#-Mq-ikLO_R3Qbjub_YfI-r8-50:-Mq_2c4--Gk5rSQ9ha3H:bhtqu7j) ([raw file](https://github.com/bible-translation-tools/orature/blob/165aff6ffb5305f2bdfdebd4e0dfe6b2fe6714b1/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt#L50)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

I disagree. All checkboxes selected should not  be the same as no checkboses selected.
</blockquote></details>

Let's take a look into a common example of how filter check boxes are used: online shopping sites like Amazon or Ebay. The check boxes are meant to refine the results with a filter criteria. It is reasonable that if there is box selected, then it will show ALL items because there is no specific constraint applied. In the case when all boxes are selected, I believe that also means "give me all of these" = ALL result.

I also brought this to Thomas earlier and got the same approval upon the behavior of the check boxes.

**Comment by AnonymousWalker** [comment]
___
*[jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModelTest.kt, line 107 at r10](https://reviewable.io/reviews/bible-translation-tools/orature/434#-MqZcb81EZR4_JC-Z8Bp:-Mq_4lP250Xb_c_Cc_Mv:b-sl72t8) ([raw file](https://github.com/bible-translation-tools/orature/blob/8258cb4802dd99b8e74b7b732c30f8a1651f9cbe/jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/HomePageViewModelTest.kt#L107)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

Perhaps there should be a method here. ```verify(mockWorkbook).someMethod()```?
</blockquote></details>

the assertEquals() line has already done the verification. 
` mockWorkbook ` is a mock object to represent the data returned and having no behavior setup to be invoked, so verifying any method of may be unnecessary.

**Review by mXaln — COMMENTED** [review]
> Reviewed 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434)* status: 12 of 13 files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker and @mXaln)

___
*[jvm/workbookapp/build.gradle, line 144 at r12](https://reviewable.io/reviews/bible-translation-tools/orature/434#-Mq_6Hy73JodQhY9qtcu:-Mq_6Hy73JodQhY9qtcv:b110l0q) ([raw file](https://github.com/bible-translation-tools/orature/blob/f8e25dc2ee94caa8444e9974a64d30f93022282a/jvm/workbookapp/build.gradle#L144)):*
> ```Gradle
>     testImplementation "org.testfx:testfx-core:$testFxVer"
>     testImplementation "org.testfx:testfx-junit:$testFxVer"
>     testImplementation "org.testfx:openjfx-monocle:jdk-11+26"
> ```

Does version 11 work on your machine? On my machine only version 12 worked,
___
*[jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt, line 50 at r8](https://reviewable.io/reviews/bible-translation-tools/orature/434#-Mq-ikLO_R3Qbjub_YfI-r8-50:-Mq_6FIh-i6MtU6gQxqC:bhgql7p) ([raw file](https://github.com/bible-translation-tools/orature/blob/165aff6ffb5305f2bdfdebd4e0dfe6b2fe6714b1/jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/LanguageSelectionViewModel.kt#L50)):*
<details><summary><i>Previously, AnonymousWalker (Tony Anh Tran) wrote…</i></summary><blockquote>

Let's take a look into a common example of how filter check boxes are used: online shopping sites like Amazon or Ebay. The check boxes are meant to refine the results with a filter criteria. It is reasonable that if there is box selected, then it will show ALL items because there is no specific constraint applied. In the case when all boxes are selected, I believe that also means "give me all of these" = ALL result.

I also brought this to Thomas earlier and got the same approval upon the behavior of the check boxes.



</blockquote></details>

All right then

**Comment by AnonymousWalker** [comment]
___
*[jvm/workbookapp/build.gradle, line 144 at r12](https://reviewable.io/reviews/bible-translation-tools/orature/434#-Mq_6Hy73JodQhY9qtcu:-Mq_sZAE8Vt932Taz-GK:b-5qlj0g) ([raw file](https://github.com/bible-translation-tools/orature/blob/f8e25dc2ee94caa8444e9974a64d30f93022282a/jvm/workbookapp/build.gradle#L144)):*
<details><summary><i>Previously, mXaln (Maxim) wrote…</i></summary><blockquote>

Does version 11 work on your machine? On my machine only version 12 worked,
</blockquote></details>

I'll try 12 then. I haven't seen any issue with local IDE, the problem only appears on CI

**Review by jsarabia — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434)* status: 7 of 13 files reviewed, 3 unresolved discussions (waiting on @AnonymousWalker and @mXaln)

___
*[jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/Utilities.kt, line 22 at r13](https://reviewable.io/reviews/bible-translation-tools/orature/434#-MsWnMjd-9yAj1oAHu-b:-MsWnMjd-9yAj1oAHu-c:bqum8fo) ([raw file](https://github.com/bible-translation-tools/orature/blob/9409549c990f7d6f95cbddf7f80bd4f171c1cda9/jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/Utilities.kt#L22)):*
> ```Kotlin
> 
> class Utilities {
>     companion object {
> ```

These don't need to be in a companion object; or a class. They can just be functions.

**Review by AnonymousWalker — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434)* status: 2 of 14 files reviewed, 3 unresolved discussions (waiting on @AnonymousWalker, @jsarabia, and @mXaln)

___
*[jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/Utilities.kt, line 22 at r13](https://reviewable.io/reviews/bible-translation-tools/orature/434#-MsWnMjd-9yAj1oAHu-b:-MtF6Ef99_4Dtk_3bVCw:b-896fix) ([raw file](https://github.com/bible-translation-tools/orature/blob/9409549c990f7d6f95cbddf7f80bd4f171c1cda9/jvm/workbookapp/src/test/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/ui/viewmodel/Utilities.kt#L22)):*
<details><summary><i>Previously, jsarabia (Joe) wrote…</i></summary><blockquote>

These don't need to be in a companion object; or a class. They can just be functions.
</blockquote></details>

Done.

**Comment by AnonymousWalker** [comment]
I will include the failed test method in a separate PR to avoid blocking this one.

```
org.wycliffeassociates.otter.jvm.workbookapp.ui.viewmodel.TranslationViewModelTest > createTranslation_showHideProgress FAILED
    java.lang.UnsatisfiedLinkError: no javafx_font_pango in java.library.path: [/usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]
```

**Review by mXaln — APPROVED** [review]
> Reviewed 3 of 11 files at r14, 1 of 6 files at r15, 2 of 3 files at r16, 11 of 12 files at r17, 1 of 1 files at r18, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434)* status: all files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)

**Review by jsarabia — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/bible-translation-tools/orature/434#-:-MtmQK-I0ORxqWXxrheV:bnfp4nl)* status: all files reviewed, 2 unresolved discussions (waiting on @AnonymousWalker and @jsarabia)

**Pull request merged**
